internal/pool/pool.go

```diff
@@ -172,16 +197,61 @@ func New[PT Item[T], T any](
 
 	p.createItem = createItemWithTimeoutHandling(p.createItem, p)
 
-	p.idle = make([]PT, 0, p.limit)
-	p.index = make(map[PT]struct{}, p.limit)
 	p.stats = &safeStats{
 		v:        stats.Stats{Limit: p.limit},
 		onChange: p.trace.OnChange,
 	}
 
+	for i := 0; i < defaultSpawnGoroutinesNumber; i++ {
+		go p.spawnItems(ctx)
+	}
+
 	return p
 }
 
+// spawnItems creates one item per each available itemToken and sends new item to internal item queue.
+// It ensures that pool would always have amount of connections equal to configured limit.
+// If item creation ended with error it will be retried infinity with configured interval until success.
+func (p *Pool[PT, T]) spawnItems(ctx context.Context) {
+spawnLoop:
+	for {
+		select {
+		case <-ctx.Done():
+			break spawnLoop
+		case <-p.done:
+			break spawnLoop
+		case <-p.itemTokens:
+			// got token, must create item
+			for {
+				item, err := p.createItem(ctx)
+				if err != nil {
+					select {
+					case <-ctx.Done():
+						break spawnLoop
+					case <-p.done:
+						break spawnLoop
+					case <-time.After(defaultCreateRetryDelay):
```

Comment:
time.After produce timer inside - this can be provokes the goroutines/rosurces leak
You can create ticker/timer before select - and call directly ticker.Stop() in defer